packages/db/prisma/extensions/base.ts

```diff
@@ -88,33 +47,54 @@ export const baseExtension = Prisma.defineExtension((prisma) =>
           }
 
           const formattedValues = addresses
+            .map(({ address }) => [address, NOW_SQL, NOW_SQL])
+            .map((rowColumns) => Prisma.sql`(${Prisma.join(rowColumns)})`);
+
+          return prisma.$executeRaw`
+            INSERT INTO address (
+              address,
+              inserted_at,
+              updated_at
+            ) VALUES ${Prisma.join(formattedValues)}
+            ON CONFLICT (address) DO UPDATE SET
+              updated_at = NOW()
+          `;
+        },
+      },
+      addressHistory: {
+        upsertMany(addressEntries: AddressHistory[]) {
+          if (!addressEntries.length) {
+            return (
+              Prisma.getExtensionContext(this) as any
+            ).zero() as PrismaPromise<ZeroOpResult>;
+          }
+
+          const formattedValues = addressEntries
             .map(
               ({
                 address,
+                category,
                 firstBlockNumberAsReceiver,
                 firstBlockNumberAsSender,
               }) => [
                 address,
-                firstBlockNumberAsSender,
+                Prisma.sql`${category.toLowerCase()}::category`,
                 firstBlockNumberAsReceiver,
-                NOW_SQL,
-                NOW_SQL,
+                firstBlockNumberAsSender,
               ]
             )
-            .map((rowColumns) => Prisma.sql`(${Prisma.join(rowColumns)})`);
+            .map((rowColumns) => Prisma.join(rowColumns, ",", "(", ")"));
```

Comment:
Why that join?

Reply:
It's a utility function that allows you to join a set of raw sql fragments using a separator and returning a `Prisma.sql` type. This  is the only type allowed by Prisma when writing raw sql queries using `executeRaw()` or `queryRaw()` prisma functions